src/OmniSharp.DotNetTest/Services/RunTestsInClassService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+using OmniSharp.DotNetTest.Models;
+using OmniSharp.Eventing;
+using OmniSharp.Mef;
+using OmniSharp.Services;
+
+namespace OmniSharp.DotNetTest.Services
+{
+    [OmniSharpHandler(OmniSharpEndpoints.V2.RunAllTestsInClass, LanguageNames.CSharp)]
+    internal class RunTestsInClassService : BaseTestService<RunTestsInClassRequest, RunTestResponse[]>
+    {
+        [ImportingConstructor]
+        public RunTestsInClassService(OmniSharpWorkspace workspace, DotNetCliService dotNetCli, IEventEmitter eventEmitter, ILoggerFactory loggerFactory)
+            : base(workspace, dotNetCli, eventEmitter, loggerFactory)
+        {
+        }
+
+        protected override RunTestResponse[] HandleRequest(RunTestsInClassRequest request, TestManager testManager)
+        {
+            List<RunTestResponse> responses = new List<RunTestResponse>();
+            if (testManager.IsConnected)
+            {
+                foreach (var methodName in request.MethodNames)
+                    responses.Add(testManager.RunTest(methodName, request.TestFrameworkName, request.TargetFrameworkVersion));
```

Comment:
Should there be a method that can run tests in bulk (taking several method names) rather than looping through the tests and running them individually?

Reply:
I would expect running several tests at once to be faster.

Reply:
Looking at what you've done with `DebugGetStartInfoAsync` in `VSTestManager`, I think you should probably do the same for `RunTest`.